crt_portal/cts_forms/views.py

```diff
@@ -255,11 +255,20 @@ def index_view(request):
     page = request.GET.get('page', 1)
 
     # Validate requested sort params
-    report_fields = [f.name for f in Report._meta.fields]
-    if all(elem.replace("-", '') in report_fields for elem in sort) is False:
+    valid_fields = [f.name for f in Report._meta.fields] + [f.name for f in EmailReportCount._meta.fields]
+    if all(elem.replace("-", '') in valid_fields for elem in sort) is False:
         raise Http404(f'Invalid sort request: {sort}')
 
-    requested_reports = report_query.order_by(*sort)
+    requested_reports = report_query.annotate(email_count=F('email_report_count__email_count'))
+
+    # apply the sort items individually so that we can push nulls to the back
+    for sort_item in sort:
+        nulls_last = 'email_count' in sort_item
+        if sort_item[0] == SORT_DESC_CHAR:
+            requested_reports = requested_reports.order_by(F(sort_item[1::]).desc(nulls_last=nulls_last))
+        else:
+            requested_reports = requested_reports.order_by(F(sort_item).asc(nulls_last=nulls_last))
```

Comment:
I don't _love_ this bit, so open to any clever alternative approaches y'all might come up with. The issue here is that Django's `nulls_last` arg has to be supplied to `asc` or `desc`, which means we have to call them explicitly rather than shorthanding (e.g. `-email_count`). So here we're unwinding the query params and pushing the sort fields into the query one by one to enable that.

Reply:
This doesn't handle multiple sort items since each order_by() call will clear any previous ordering. [ref](https://docs.djangoproject.com/en/2.2/ref/models/querysets/#order-by)
As-is, `sort = ['status', '-email_count']` would only order by `email_count`

The current UI only allows folks to sort by a single field, let's handle just that case for now.

Reply:
🤦‍♂️ right you are, thanks! I was thinking the `order_bys` stacked. This could be modified slightly to instead build up a list of query expressions to be supplied to `order_by` if/when we want to support multiple sort items. Perhaps we should do that now, since leaving this as it is makes it appear that we support it on the backend when in reality we don't

Reply:
Going to fix this so that we actually support multiple sorts on the backend, stand by